kpi/views/v1/import_task.py

```diff
@@ -59,6 +62,53 @@ def create(self, request, *args, **kwargs):
                 'importtask-detail',
                 kwargs={'uid': import_task.uid},
                 request=request),
-            'status': ImportTask.PROCESSING
+            'status': ImportTask.PROCESSING,
+            'default_project_name': default_project_name
         }, status.HTTP_201_CREATED)
 
+    @staticmethod
+    def _get_filenames(
+        request: Request,
+    ) -> Union[Tuple[str, str], Tuple[None, None]]:
+        """
+        Return a tuple of the submitted file or url-linked file's name with
+        and without its extension
+
+        Args:
+            request (Request): DRF request object
+
+        Returns:
+            Tuple[str, str] or Tuple[None, None]: A tuple containing first the
+            filename with its extension and secondly the filename without its
+            extension. If there is no filename, a tuple of None values is
+            returned
+        """
+
+        XLS_FILE_PATTERN = '((.*?)\.xlsx?)'
+        XLS_URL_PATTERN = '\w+="?((.*?)\.xlsx?)"?'
```

Comment:
I don't see when your `XLS_URL_PATTERN` can match. 
Do you have an example of which URL you used?

Reply:
@noliveleger, if you run `curl --head --location <url>` then you get this in the `content-disposition` response header:

```
content-disposition: attachment; filename="BasicChoicesExample.xlsx"; filename*=UTF-8''Basic%20Choices%20Example.xlsx
```

So using `XLS_URL_PATTERN` will match the first filename.

Reply:
It seems like `Content-Disposition` headers are a pretty varied species: https://stackoverflow.com/q/37060344/2402324
What would you think about using a library like [rfc6266](https://pypi.org/project/rfc6266/) to handle parsing them?

Reply:
Good idea 👍 I'll test that out.

Reply:
@jnm, [rfc6266](https://github.com/g2p/rfc6266) seems to be dead... and it also doesn't seem to work 😂

Reply:
@joshuaberetta looks like there are some alternatives
![image](https://user-images.githubusercontent.com/2085013/107438945-54645b80-6aff-11eb-939b-274eb0aee389.png)

Reply:
whoops, i had high hopes for forks that were updated in 2019 (vs. 2013), but they don't work either. there's a helpful comment on one of them, though:
> You might consider [werkzeug.http.parse_options_header](https://werkzeug.palletsprojects.com/en/0.15.x/http/#werkzeug.http.parse_options_header)

https://github.com/SWW13/python-rfc6266-parser/issues/2#issuecomment-482951496

we already have werkzeug as a development dependency; this would require moving it to the common `requirements.in` file.

demonstration:
```
In [1]: from werkzeug.http import parse_options_header                                                                               

In [2]: import requests                                                                                                              

In [3]: resp = requests.get('https://docs.google.com/spreadsheets/d/e/2PACX-1vTuRyLRmTgFZrG1sJWjkurd0GJxH74YiA518-KOhgWk0LxVm2NcWkQP0
   ...: iQjgHTqnCVqFOgPiGWBPSBT/pub?output=xlsx')                                                                                    

In [4]: parse_options_header(resp.headers['content-disposition'])                                                                    
Out[4]: ('attachment', {'filename': 'لا يستخدم الجميع الأحرف اللاتينية.xlsx'})
```